### ISSUE TITLE
Custom Loss Function Support

### ISSUE THREAD

**Issue by scorpionhiccup** [facebookresearch/fastText#679]
I was wondering if Fasttext supports the addition of custom loss functions?

If so, pls. guide as to how it can be done.

If not, pls. let us know as to why that is the case..

**Comment by Celebio** [comment]
Hi @scorpionhiccup ,
No it is not supported. We can think about something like this in the future, but we can't commit ourselves.

For the moment you have to implement it by adding the corresponding function in model.cc. You have to check every time `args_->loss` is accessed and follow the `update` function.

Please note that if the loss function you are interested in is multilabel, we already implemented it but we are doing further tests and experiments before publishing it, so waiting would be enough for this one.

Regards,
Onur

**Comment by Celebio** [comment]
Hi @scorpionhiccup ,
We [published a new loss function](8850c51b972ed68642a15c17fbcd4dd58766291d) that computes a cross-entropy loss for each unit of the output independently, and does a summation.

We are also working on the ability to allow the user to provide a custom loss function easily, but we can't give you an ETA for the moment.

I will close this issue, please feel free to re-open it if you have further questions.

Best regards,
Onur
